spigot_v1_15_R2/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/Spigot_v1_15_R2.java

```diff
@@ -318,6 +320,42 @@ public BaseBlock getBlock(Location location) {
         return state.toBaseBlock();
     }
 
+    private void updateNeighbours(World world, BlockPosition blockposition, Chunk chunk, IBlockData oldBlock, IBlockData newBlock, IBlockData actualBlock, int i) {
+        if (actualBlock == newBlock) {
+            if (oldBlock != actualBlock) {
+                world.b(blockposition, oldBlock, actualBlock);
+            }
+
+            if ((i & 2) != 0 && (!world.isClientSide || (i & 4) == 0) && (world.isClientSide || chunk == null || chunk.getState() != null && chunk.getState().isAtLeast(
+                    PlayerChunk.State.TICKING))) {
+                ((ChunkProviderServer) world.getChunkProvider()).flagDirty(blockposition);
+            }
+
+            if (!world.isClientSide && (i & 1) != 0) {
+                world.getType(blockposition.west()).doPhysics(world, blockposition.west(), oldBlock.getBlock(), blockposition, false);
+                world.getType(blockposition.east()).doPhysics(world, blockposition.east(), oldBlock.getBlock(), blockposition, false);
+                world.getType(blockposition.down()).doPhysics(world, blockposition.down(), oldBlock.getBlock(), blockposition, false);
+                world.getType(blockposition.up()).doPhysics(world, blockposition.up(), oldBlock.getBlock(), blockposition, false);
+                world.getType(blockposition.north()).doPhysics(world, blockposition.north(), oldBlock.getBlock(), blockposition, false);
+                world.getType(blockposition.south()).doPhysics(world, blockposition.south(), oldBlock.getBlock(), blockposition, false);
+
+                if (newBlock.isComplexRedstone()) {
+                    world.updateAdjacentComparators(blockposition, newBlock.getBlock());
+                }
+            }
+
+            if ((i & 16) == 0) {
+                int j = i & -2;
+                oldBlock.b(world, blockposition, j);
```

Comment:
A comment on what these three methods are doing would be nice, unless this is all copied from somewhere, in which case that place should be referenced.